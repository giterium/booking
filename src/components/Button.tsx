import React  from 'react';

interface ButtonTypes {
    title: string,
    onClick?: Function,
    className?: string,
}

export const Button = ( props : ButtonTypes) => (
    <button onClick={props.onClick} className={props.className}>{props.title}</button>
);