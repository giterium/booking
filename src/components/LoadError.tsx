import React  from 'react';

export const LoadError = () => (
    <div><br /><br /><p>Извините! Ошибка при загрузке...</p></div>
);