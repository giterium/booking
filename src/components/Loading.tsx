import React  from 'react';
export const Loading = () => <div style={{textAlign: 'center', marginTop: '200px'}}><br /><br /><br />Loading...</div>;