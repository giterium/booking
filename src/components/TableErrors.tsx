import React  from 'react';

export type TableErrorsProps = {
    errors: string
};

export const TableErrors = ({ errors }: TableErrorsProps) =>
    <tr>
        <td colSpan={2}>
            <div className="errorBox">{errors && (<div className="invalid-feedback-box">{errors}</div>)}</div>
        </td>
    </tr>
;