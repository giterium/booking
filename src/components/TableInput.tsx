import React  from 'react';

interface TableInputTypes {
    title: string,
    type?: string,
    className: string,
    name: string,
    changeUpdate: (e: any) => void,
    onKeyPress?: (e: EventTarget) => void,
    onFocus?: (e: EventTarget) => void,
    onClick?: (e: EventTarget) => void,
    defaultValue: string | number
}

export const TableInput = ( props : TableInputTypes) => (
    <tr>
        <td className="updateFieldTitle">{props.title}:</td>
        <td>
            <input
                type={(typeof props.type == 'undefined') ? 'text': props.type}
                className={(typeof props.className == 'undefined') ? '': props.className}
                name={props.name}
                onChange={props.changeUpdate}
                onFocus={props.onFocus}
                onClick={props.onClick}
                onKeyUp={props.changeUpdate}
                onKeyPress={props.onKeyPress}
                defaultValue={props.defaultValue}
            />
        </td>
    </tr>
);