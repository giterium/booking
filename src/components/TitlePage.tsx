import React from "react";

type TitleProps = { title: string };
export const TitlePage = ({ title }: TitleProps) => <div className="titlePage">{title}</div>;