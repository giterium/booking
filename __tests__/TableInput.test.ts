import React  from 'react';
import { TableInput } from "../src/components/TableInput";
const setUp = (props) => shallow(<TableInput {...props} />);

describe("should render TableInput component", () => {
    let component = '';
    beforeEach(() => {
        component = setUp();
    });

    it("should render TableInput component", () => {
        expect(component.debug()).toMatchSnapshot();
    });

    it("should call changeUpdate method", () => {
        const mockCallBack = jest.fn();
        const component = shallow(<TableInput changeUpdate={mockCallBack} />);
        expect(mockCallBack.mock.calls.length).toBe(0);
        component.find("input").simulate("change");
        expect(mockCallBack.mock.calls.length).toBe(1);
    });

    it("should call changeUpdate with increased value", () => {
        const mockCallBack = jest.fn();
        const component = shallow(<TableInput changeUpdate={mockCallBack} />);
        expect(mockCallBack.mock.calls.length).toBe(0);
        component.find('input').simulate('change', 'the-value');
        expect(mockCallBack).toHaveBeenCalledWith( 'the-value');
    });

    it("should use className", () => {
        component = setUp({ className: 'class' });
        expect(component.find('input').hasClass('class')).toEqual(true)
    });

    it("should use defaultValue", () => {
        component = setUp({ defaultValue: 'Hello' });
        const input = component.find('input');
        expect(input.prop('defaultValue')).toEqual('Hello');
    });
});