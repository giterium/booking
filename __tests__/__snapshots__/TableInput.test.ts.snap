// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TableInput component should render TableInput component 1`] = `
"<tr>
  <td className=\\"updateFieldTitle\\">
    :
  </td>
  <td>
    <input type=\\"text\\" className=\\"\\" name={[undefined]} onChange={[undefined]} onFocus={[undefined]} onClick={[undefined]} onKeyUp={[undefined]} onKeyPress={[undefined]} defaultValue={[undefined]} />
  </td>
</tr>"
`;
